src/core/Ros.js

```diff
@@ -56,7 +57,17 @@ Ros.prototype.__proto__ = EventEmitter2.prototype;
  * @param url - WebSocket URL for Rosbridge
  */
 Ros.prototype.connect = function(url) {
-  this.socket = assign(new WebSocket(url), socketAdapter(this));
+  if( typeof(io) !== 'undefined' && this.socketio){
```

Comment:
This would look like `if (this.transportLibrary === 'socket.io') {` ; it will blow up if `io` is not defined, but requested as a transport library.